DropDown width issues solved

boutinb opened the pull request:
Check test DropDown in jaspTestModule

The fieldWidth property gives the width of the internal dropdown control. If this property is set, it fixes the width. This feature is needed, especially when DropDown controls are used in a ComponentList, where the width of the controls should not changed with the values of the dropdown.

This is also needed to solves some problem in PR https://github.com/jasp-stats/jaspProcess/pull/108

boutinb committed:
DropDown width issues solved

boutinb committed:
Replace controlMinWidth by fieldWidth in VariablesForm

boutinb committed:
Add indicator to calculate dropdown width

JorisGoosen commented on QMLComponents/components/JASP/Controls/ComboBox.qml:
```diff
@@ -19,59 +19,19 @@ ComboBoxBase
 	property alias	indexDefaultValue:		comboBox.currentIndex
-	property alias	fieldWidth:				control.width
+	property alias	fieldWidth:				control.implicitWidth
+	property alias  controlText:			controlText
```
```
	property alias  controlText:			controlText
```
???

boutinb replied:
I forgot to remove this

JorisGoosen commented on QMLComponents/components/JASP/Controls/ComboBox.qml:
```diff
@@ -19,59 +19,19 @@ ComboBoxBase
 	property alias	indexDefaultValue:		comboBox.currentIndex
-	property alias	fieldWidth:				control.width
+	property alias	fieldWidth:				control.implicitWidth
+	property alias  controlText:			controlText
```
```suggestion
	property alias  controlText:			control.text
```

Maybe you mean this?

boutinb replied:
No this property must be removed. `value` is already the value of the DropDown.

JorisGoosen replied:
So remove it then

JorisGoosen commented on QMLComponents/controls/comboboxbase.h:
```diff
@@ -80,10 +80,10 @@ protected slots:
 
 protected:
 	bool _checkLevelsConstraints()			override;
+	void _setLongestValue();
```
???
```suggestion
```

boutinb committed:
Fix DropDown initialization in MetaAnalysis

In the Effect Size Computation of the Meta Analysis module, the dropdowns in a ComponentList depends on each other for their values.
TO solve this, just use the _unusedInitialValue.

JorisGoosen reviewed (changes requested):
You see them now then?

JorisGoosen committed:
set height of popup otherwise it doesnt get moved up on too big for screen

boutinb committed:
Update QMLComponents/controls/comboboxbase.h

Co-authored-by: Joris Goosen <Joris@JorisGoosen.nl>

JorisGoosen committed:
ok so padding needs to be included in the implicitsize of the popup

boutinb committed:
Remove controlText alias

shun2wang reviewed (approved):
DropDown now works well for me.

boutinb commented:
Should we merge it @JorisGoosen ?
